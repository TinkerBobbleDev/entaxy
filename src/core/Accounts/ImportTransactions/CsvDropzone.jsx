import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import Dropzone from 'react-dropzone'
import Icon from '@mdi/react'
import { mdiFileUploadOutline } from '@mdi/js'

const styles = theme => ({
  root: {
    border: '3px dashed rgb(200, 200, 200)',
    borderRadius: '5px',
    padding: '50px',
    cursor: 'pointer',
    display: 'flex',
    'align-items': 'center',
    'justify-content': 'center',
    'flex-direction': 'row'
  },
  menuIcon: {
    marginRight: '5px',
    'vertical-align': 'bottom',
    fill: theme.palette.text.secondary
  }
})

const CsvDropzone = ({
  classes,
  handleFileUpload,
  file,
  error
}) => (
  <Dropzone multiple={false} className={classes.root} onDrop={handleFileUpload}>
    <div>
      {file &&
        <Typography variant="subtitle2" align="center">
          <Icon
            path={mdiFileUploadOutline}
            size={1}
            className={classes.menuIcon}
            color="textSecondary"
          />
          {file.name}
        </Typography>
      }
      {error &&
        <Typography variant="subtitle2" align="center" color="error">{error}</Typography>
      }
      {!file &&
        <div>
          <Typography variant="h4" align="center" color="textSecondary">
            Drag a CSV file here
          </Typography>
          <Typography variant="subtitle2" align="center" color="textSecondary">
            or click to select a file to upload.
          </Typography>
        </div>
      }
    </div>
  </Dropzone>
)

CsvDropzone.propTypes = {
  classes: PropTypes.object.isRequired,
  handleFileUpload: PropTypes.func.isRequired,
  error: PropTypes.string,
  file: PropTypes.object
}

CsvDropzone.defaultProps = {
  error: null,
  file: null
}

export default withStyles(styles)(CsvDropzone)
