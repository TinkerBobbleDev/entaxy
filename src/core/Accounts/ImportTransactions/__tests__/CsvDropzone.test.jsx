import React from 'react'
import renderer from 'react-test-renderer'
import CsvDropzone from '../CsvDropzone'

describe('CsvDropzone', () => {
  const mockHandleFileUpload = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot without a file', () => {
    const component = renderer.create((
      <CsvDropzone
        handleFileUpload={mockHandleFileUpload}
        classes={{ }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with a good file', () => {
    const component = renderer.create((
      <CsvDropzone
        handleFileUpload={mockHandleFileUpload}
        file={{ }}
        classes={{ }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with a bad file', () => {
    const component = renderer.create((
      <CsvDropzone
        handleFileUpload={mockHandleFileUpload}
        file={{ }}
        error="Some error"
        classes={{ }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
