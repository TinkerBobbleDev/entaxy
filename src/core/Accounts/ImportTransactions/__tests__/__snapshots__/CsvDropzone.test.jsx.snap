// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsvDropzone matches snapshot with a bad file 1`] = `
<div
  aria-disabled={false}
  className="CsvDropzone-root-1"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div>
    <h6
      className="MuiTypography-root-3 MuiTypography-subtitle2-22 MuiTypography-alignCenter-26"
    >
      <svg
        className="CsvDropzone-menuIcon-2"
        style={
          Object {
            "width": "1.5rem",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M14,2L20,8V20C20,21.1 19.1,22 18,22H6C4.9,22 4,21.1 4,20V4C4,2.9 4.9,2 6,2H14M18,20V9H13V4H6V20H18M12,12L16,16H13.5V19H10.5V16H8L12,12Z"
          style={
            Object {
              "fill": "textSecondary",
            }
          }
        />
      </svg>
    </h6>
    <h6
      className="MuiTypography-root-3 MuiTypography-subtitle2-22 MuiTypography-colorError-37 MuiTypography-alignCenter-26"
    >
      Some error
    </h6>
  </div>
  <input
    autoComplete="off"
    disabled={false}
    multiple={false}
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;

exports[`CsvDropzone matches snapshot with a good file 1`] = `
<div
  aria-disabled={false}
  className="CsvDropzone-root-1"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div>
    <h6
      className="MuiTypography-root-3 MuiTypography-subtitle2-22 MuiTypography-alignCenter-26"
    >
      <svg
        className="CsvDropzone-menuIcon-2"
        style={
          Object {
            "width": "1.5rem",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M14,2L20,8V20C20,21.1 19.1,22 18,22H6C4.9,22 4,21.1 4,20V4C4,2.9 4.9,2 6,2H14M18,20V9H13V4H6V20H18M12,12L16,16H13.5V19H10.5V16H8L12,12Z"
          style={
            Object {
              "fill": "textSecondary",
            }
          }
        />
      </svg>
    </h6>
  </div>
  <input
    autoComplete="off"
    disabled={false}
    multiple={false}
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;

exports[`CsvDropzone matches snapshot without a file 1`] = `
<div
  aria-disabled={false}
  className="CsvDropzone-root-1"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div>
    <div>
      <h4
        className="MuiTypography-root-3 MuiTypography-h4-18 MuiTypography-colorTextSecondary-36 MuiTypography-alignCenter-26"
      >
        Drag a CSV file here
      </h4>
      <h6
        className="MuiTypography-root-3 MuiTypography-subtitle2-22 MuiTypography-colorTextSecondary-36 MuiTypography-alignCenter-26"
      >
        or click to select a file to upload.
      </h6>
    </div>
  </div>
  <input
    autoComplete="off"
    disabled={false}
    multiple={false}
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;
